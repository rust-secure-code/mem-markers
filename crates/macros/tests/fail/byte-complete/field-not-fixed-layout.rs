use mem_markers::ByteComplete;

#[derive(ByteComplete)]
struct Foo {
    a: Other,
}

struct Other {}
fn main() {}
