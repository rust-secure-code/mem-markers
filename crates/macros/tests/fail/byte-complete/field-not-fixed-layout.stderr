error[E0277]: the trait bound `Other: mem_markers::byte_complete::ByteComplete` is not satisfied
 --> $DIR/field-not-fixed-layout.rs:5:8
  |
3 | #[derive(ByteComplete)]
  |          ------------ required by this bound in `FooImplsByteComplete`
4 | struct Foo {
  |        ---
5 |     a: Other,
  |        ^^^^^ the trait `mem_markers::byte_complete::ByteComplete` is not implemented for `Other`
